cpp/periodic_module/periodic.cpp

```diff
@@ -203,28 +176,22 @@ std::string ConstructFinalQuery(const std::string &running_query, const std::str
   return fmt::format("{} {}", prefix_query, running_query);
 }
 
-void ExecuteRunningQuery(const std::string running_query, const std::vector<std::string> &columns,
-                         const std::vector<std::vector<mg::Value>> &batch) {
-  if (!batch.size()) {
+void ExecuteRunningQuery(const mgp::QueryExecution &query_execution, std::string running_query,
+                         const mgp::ExecutionHeaders &headers, const mgp::List &batch) {
+  if (batch.Empty()) {
     return;
   }
 
-  auto param_names = ExtractParamNames(columns, batch[0]);
+  auto param_names = ExtractParamNames(headers, batch[0].ValueList());
   auto prefix_query = ConstructQueryPrefix(param_names);
   auto final_query = ConstructFinalQuery(running_query, prefix_query);
 
-  auto query_params = ConstructQueryParams(columns, batch);
+  auto query_params = ConstructQueryParams(headers, batch);
 
-  mg::Client::Params session_params{.host = "localhost", .port = 7687};
-  auto client = mg::Client::Connect(session_params);
-  if (!client) {
-    throw std::runtime_error("Unable to connect to client!");
-  }
-  if (!client->Execute(final_query, query_params.AsConstMap())) {
-    throw std::runtime_error("Error while executing periodic iterate!");
-  }
+  auto execution_result = query_execution.ExecuteQuery(final_query, query_params);
 
-  client->DiscardAll();
+  while (execution_result.PullOne()) {
+  }
```

Comment:
The role of this empty loop is to pull all results, right?

Reply:
yes